Tests/Get-FileHardLink.Tests.ps1

```diff
@@ -0,0 +1,155 @@
+            {
+                if( (Test-Path -Path $linkPath -PathType Leaf) )
+                {
+                    Write-Verbose -Message ('Removing "{0}": this file exists.' -f $linkPath)
+                    Remove-Item -Path $linkPath
+                }
+
+                if( -not (Test-Path -Path $linkPath -PathType Leaf) )
+                {
+                    Write-Verbose ('Creating hardlink "{0}" -> "{1}".' -f $linkPath,$targetPath)
+                    New-Item -ItemType HardLink -Path $linkPath -Value $targetPath
+                }
+            }
+        }
+        finally
+        {
+            Pop-Location
+        }
+    }
+
+    function GivenFile
+    {
+        param(
+            [Parameter(Mandatory)]
+            [String] $Path,
+
+            [String] $In
+        )
+
+        if( $In )
+        {
+            $Path = Join-Path -Path $In -ChildPath $Path
+        }
+
+        New-Item -Path $Path -ItemType 'File'
+    }
+
+    function ThenFailed
+    {
+        param(
+            [Parameter(Mandatory)]
+            [string] $WithErrorMatching
+        )
+
+        $Global:Error[-1] | Should -Match $WithErrorMatching
+    }
+
+    function ThenFile
+    {
+        param(
+            [Parameter(Mandatory)]
+            [String]$Path,
+
+            [switch]$Not,
+
+            [switch]$Exists,
+
+            [Object]$HasLinkType,
+
+            [Object]$Targets
+        )
+
+        $fullPath = Join-Path -Path $testRoot -ChildPath $Path
+
+        if( $Exists )
+        {
+            $fullPath | Should -Not:$Not -Exist
+        }
+
+        if( $PSBoundParameters.ContainsKey('HasLinkType') )
+        {
+            Get-Item -Path $fullPath -Force | Select-Object -Expand 'LinkType' | Should -Not:$Not -Be $HasLinkType
+        }
+
+        if( $PSBoundParameters.ContainsKey('Targets') )
+        {
+            if( $Targets -and -not [IO.Path]::IsPathRooted($Targets) )
+            {
+                $Targets = Join-Path -Path $testRoot -ChildPath $Targets -Resolve
+            }
+            Get-FileHardLink -Path $fullPath | Should -Not:$Not -Contain $Targets
+        }
+    }
+}
+
+Describe 'Get-FileHardLink' {
+    BeforeEach { 
+        $script:testRoot = $null
+        $script:failed = $false
+        $script:testRoot = Join-Path -Path $TestDrive -ChildPath ($script:testNum++)
+        New-Item -Path $script:testRoot -ItemType 'Directory'
+        $Global:Error.Clear()
+    }
+
+    It 'should fail when there is no driver file' {
+        $linkPath = Join-Path -Path '.' -ChildPath '.kitchen.local.yml'
+        GivenHardlink -LinkPath $linkPath -ThatTargets 'testDriver' -ErrorAction SilentlyContinue
+        ThenFailed -WithErrorMatching '\.kitchen\.testDriver\.yml" does not exist'
+        ThenFile '.kitchen.local.yml' -Not -Exists
+    }
+
+    It 'should retrieve targets from link when there is a driver file' {
+        $linkPath = @()
+        $linkPath += Join-Path -Path '.' -ChildPath '.kitchen.local.yml'

```

Comment:
Use relative paths. Change the file names so they are specific to this test. Weird to see specific file names like that in this test fixture.